pkg/issuer/venafi/client/request.go

```diff
@@ -49,6 +50,31 @@ func (v *Venafi) RequestCertificate(csrPEM []byte, duration time.Duration, custo
 	if err != nil {
 		return "", err
 	}
+
+	// If the connector is TPP, we unconditionally reset any prior failed enrollment
+	// so that we don't get stuck with "Fix any errors, and then click Retry."
+	// (60% of the time) or "WebSDK CertRequest" (40% of the time).
+	//
+	// It would be preferable to only reset when necessary to avoid the extra
+	// call. We tried that in https://github.com/Venafi/vcert/pull/269. It turns
+	// out that calling "request" followed by "reset(restart=true)" causes a
+	// race in TPP.
+	//
+	// Unconditionally resetting isn't optimal, but "reset(restart=false)" is
+	// lightweight. We haven't verified that it doesn't slow things down on
+	// large TPP instances.
+	//
+	// Note that resetting won't affect the existing certificate if one was
+	// already issued.
+	tppConnector, isTPP := v.vcertClient.(*tpp.Connector)
+	if isTPP {
+		err := tppConnector.ResetCertificate(vreq, false)
+		notFoundErr := &tpp.ErrCertNotFound{}
+		if err != nil && !errors.As(err, &notFoundErr) {
+			return "", err
+		}
```

Comment:
suggestion: Haven't tested the syntax here, but is there a reason we're using `As` here? Would `Is` not be appropriate?

```suggestion
		if err != nil && !errors.Is(err, &tpp.ErrCertNotFound{}) {
			return "", err
		}
```

Reply:
If the Err field in ErrCertNotFound does not match the empty field in the provided `&tpp.ErrCertNotFound{}` value, errors.Is will return false. For that reason, we have to use errors.As() here, so we only match on the error type and not on the contents of the error.

Reply:
Fair enough, I assumed (without checking, my bad) that ErrCertNotFound was a sentinel value!